Bricks/template/decay.h

```diff
@@ -72,8 +72,8 @@ struct rmref_impl<std::tuple<TS...>&&> {
   typedef metaprogramming::map<impl, std::tuple<TS...>> type;
 };
 
-template <typename... TS>
-using rmref = typename rmref_impl<TS...>::type;
+template <typename T>
+using rmref = typename rmref_impl<T>::type;
```

Comment:
I'm not sure, but does variadic implementation above make sense with this change?